src/pybind/mgr/cephadm/module.py

```diff
@@ -2360,31 +2363,52 @@ def remove_osds(self, osd_ids: List[str],
         """
         Takes a list of OSDs and schedules them for removal.
         The function that takes care of the actual removal is
-        _remove_osds_bg().
+        process_removal_queue().
         """
 
-        daemons = self.cache.get_daemons_by_service('osd')
-        found: Set[OSDRemoval] = set()
+        daemons: List[orchestrator.DaemonDescription] = self.cache.get_daemons_by_service('osd')
+        to_remove_daemons = list()
```

Comment:
```suggestion
        to_remove_daemons = [d for d in daemons if d.daemon_id in osd_ids]
```

(just was just for my own understanding. Nothing you need to apply.)